images/macos/provision/utils/utils.sh

```diff
@@ -108,4 +108,66 @@ brew_cask_install_ignoring_sha256() {
     pushd $CASK_DIR
     git checkout HEAD -- "$TOOL_NAME.rb"
     popd
-}
\ No newline at end of file
+}
+
+get_brew_os_keyword() {
+    if is_HighSierra; then
+        echo "high_sierra"
+    elif is_Mojave; then
+        echo "mojave"
+    elif is_Catalina; then
+        echo "catalina"
+    elif is_BigSur; then
+        echo "big_sur"
+    else
+        echo "null"
+    fi
+}
+
+should_build_from_source() {
+    local tool_name=$1
+    local os_name=$2
+    local tool_info=$(brew info --json=v1 $tool_name)
+    local bottle_disabled=$(echo "$tool_info" | jq ".[0].bottle_disabled")
+
+    # No need to build from source if a bottle is disabled
+    # Use the simple 'brew install' command to download a package
+    if $bottle_disabled; then
+        echo "false"
```

Comment:
Should we use `return` after echo? to avoid running function

Reply:
Good catch, thanks!